Buildings/Controls/OBC/ASHRAE/PrimarySystem/BoilerPlant/Controller.mo

```diff
@@ -0,0 +1,2260 @@
+    "Primary pump speed regulation method"
+    annotation (Dialog(group="Boiler plant configuration parameters", enable=have_varPriPum));
+
+  parameter Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Types.SecondaryPumpSpeedControlTypes
+    speConTypSec = Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Types.SecondaryPumpSpeedControlTypes.remoteDP
+    "Secondary pump speed regulation method"
+    annotation (Dialog(group="Boiler plant configuration parameters", enable=have_varSecPum));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uBoiAva[nBoi]
+    "Boiler availability status signal"
+    annotation (Placement(transformation(extent={{-440,50},{-400,90}}),
+      iconTransformation(extent={{-140,60},{-100,100}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uBoi[nBoi]
+    "Boiler status vector from plant"
+    annotation (Placement(transformation(extent={{-440,-380},{-400,-340}}),
+      iconTransformation(extent={{-140,-190},{-100,-150}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uPriPum[nPumPri]
+    "Primary pump proven on signal from plant"
+    annotation (Placement(transformation(extent={{-440,-420},{-400,-380}}),
+      iconTransformation(extent={{-140,-220},{-100,-180}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uSecPum[nPumSec] if not have_priOnl
+    "Secondary pump proven on signal"
+    annotation (Placement(transformation(extent={{-440,-460},{-400,-420}}),
+      iconTransformation(extent={{-140,-250},{-100,-210}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.IntegerInput plaReq
+    "Plant requests"
+    annotation (Placement(transformation(extent={{-440,330},{-400,370}}),
+      iconTransformation(extent={{-140,270},{-100,310}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.IntegerInput TSupResReq
+    "Hot water supply temperature reset requests"
+    annotation (Placement(transformation(extent={{-440,360},{-400,400}}),
+      iconTransformation(extent={{-140,300},{-100,340}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TOut(
+    final unit="K",
+    displayUnit="degC",
+    final quantity="ThermodynamicTemperature")
+    "Measured outdoor air temperature"
+    annotation (Placement(transformation(extent={{-440,290},{-400,330}}),
+      iconTransformation(extent={{-140,240},{-100,280}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TSupPri(
+    final unit="K",
+    displayUnit="degC",
+    final quantity="ThermodynamicTemperature")
+    "Measured hot water supply temperature"
+    annotation (Placement(transformation(extent={{-440,250},{-400,290}}),
+      iconTransformation(extent={{-140,210},{-100,250}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TRetPri(
+    final unit="K",
+    displayUnit="degC",
+    final quantity="ThermodynamicTemperature")
+    "Measured hot water primary return temperature"
```

Comment:
Missing Boolean condition:

  > HW return temperature is required for primary-only plants and primary-secondary plants with non-condensing boilers.